src/utils.ts

```diff
@@ -2082,21 +2184,139 @@ export class newTooltip {
   }
 }
 
-export class newAxis {
+const DASH_SELECTION_WINDOW = [7, 3];
+export class SelectionBox extends newRect {
+  public minVertex: Vertex;
+  public maxVertex: Vertex;
+
+  private _previousMin: Vertex;
+  private _previousMax: Vertex;
+
+  public leftUpdate: boolean = false;
+  public rightUpdate: boolean = false;
+  public upUpdate: boolean = false;
+  public downUpdate: boolean = false;
+  constructor(
+    public origin: Vertex = new Vertex(0, 0),
+    public size: Vertex = new Vertex(0, 0)
+  ) {
+    super(origin, size);
+    this.dashLine = DASH_SELECTION_WINDOW;
+    this.selectedStyle = this.clickedStyle = this.hoverStyle = this.fillStyle = "hsla(0, 0%, 100%, 0)";
+    this.lineWidth = 0.5
+  }
+
+  get isDefined(): boolean { return (this.minVertex != undefined && this.maxVertex != undefined) }
```

Comment:
Prefer using null, by defaulting them to it

Reply:
Done